SNIPS/snip-86.md

```diff
@@ -0,0 +1,601 @@
+        "Message": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": [
+                  "version",
+                  "address",
+                  "statement",
+                  "uri",
+                  "nonce",
+                  "issuedAt",
+                  "validUntil",
+                  "validAfter"
+                ],
+                "errorMessage": "Name must be one of 'version', 'address', 'statement', 'uri', 'nonce', 'issuedAt', 'validUntil', 'validAfter'"
+              },
+              "type": {
+                "type": "string",
+                "enum": ["string", "felt"],
+                "errorMessage": "Type must be either 'string' or 'felt'"
+              }
+            },
+            "required": ["name", "type"],
+            "additionalProperties": false,
+            "errorMessage": "Items must include name and type"
+          },
+          "minItems": 6,
+          "maxItems": 8,
+          "uniqueItems": true,
+          "errorMessage": "Message must contain min 6-8 unique items "
+        },
+        "StarknetDomain": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "properties": {
+              "name": {
+                "type": "string",
+                "enum": ["name", "chainId", "version", "revision"],
+                "errorMessage": "Name must be one of 'name', 'chainId', 'version', 'revision'"
+              },
+              "type": {
+                "type": "string",
+                "enum": ["felt", "string"],
+                "errorMessage": "Type must be 'felt' or 'string'"
+              }
+            },
+            "required": ["name", "type"],
+            "additionalProperties": false,
+            "errorMessage": "Items must include name and type"
+          },
+          "minItems": 4,
+          "maxItems": 4,
+          "uniqueItems": true,
+          "errorMessage": "StarknetDomain must contain exactly 4 unique items"
+        }
+      },
+      "required": ["Message", "StarknetDomain"],
+      "additionalProperties": false,
+      "errorMessage": "Types must include Message and StarknetDomain"
+    }
+  },
+  "required": ["domain", "message", "primaryType", "types"],
+  "additionalProperties": false,
+  "errorMessage": "Data must include domain, message, primaryType, types"
+}
+```
+
+#### Message Fields
+
+This specification defines the following SIWS Message fields that can be parsed from a SIWS Message by following the rules in [Structured JSON Message Format](#structured-json-message-format).
+
+- `domain` REQUIRED. An object containing information about the domain requesting the signing.
+
+  - `chainId` REQUIRED. The Starknet network identifier. Its value MUST be either "SN_SEPOLIA" or "SN_MAIN".
+  - `name` REQUIRED. The name of the application requesting the signing.
```

Comment:
IMO the domain name should be alway the same so wallets can recognize it and display it in a nice way, the app url can be part of the message, app name doesn’t need to be signed